tests/Contrib/Unit/ZipkinToNewrelicExporterTest.php

```diff
@@ -105,7 +107,7 @@ public function shouldThrowExceptionIfInvalidDsnIsPassed(string $invalidDsn)
     {
         $this->expectException(InvalidArgumentException::class);
 
-        new Exporter('test.zipkinToNR', $invalidDsn, '');
+        new Exporter('test.zipkinToNR', $invalidDsn, '', new Client(), new HttpFactory(), new HttpFactory());
```

Comment:
This is a shift in UX, we might make a subsequent PR where we fix it by doing some heuristic.

Reply:
How do you mean @dkarlovi, how can we do this?

Reply:
Something like:

```php
    public function __construct(
        $name,
        string $endpointUrl,
        string $licenseKey,
        ClientInterface $client = null,
        RequestFactoryInterface $requestFactory = null,
        StreamFactoryInterface $streamFactory = null,
        SpanConverter $spanConverter = null,
```

And then

```php
if ($client === null) {
    if (class_exists(\Http\Discovery\HttpClientDiscovery::class)) {
        $client = \Http\Discovery\HttpClientDiscovery::find();
    } elseif (class_exists(\GuzzleHttp\Client)) {
        $client = new \GuzzleHttp\Client(); // this allows for some BC
    } else {
        throw new LogicException('No HTTP client found or provided');
    }
}
```

[A similar example](https://php-http.readthedocs.io/en/latest/discovery.html#psr-17-factory-discovery) would be there for factories. I'd have this in a trait which all the exporters could then use to simplify implementation.

Reply:
This looks like something interesting to check out, Looking forward to the next iteration. Thanks @dkarlovi